src/main.d

```diff
@@ -14,26 +14,39 @@ int main(string[] args)
 	// Check for HOME environment variable
 	if (environment.get("HOME") != ""){
 		// Use HOME environment variable
+		log.vdebug("homePath: HOME environment variable set");
 		homePath = environment.get("HOME");
 	} else {
 		if ((environment.get("SHELL") == "") && (environment.get("USER") == "")){
 			// No shell is set or username - observed case when running as systemd service under CentOS 7.x
```

Comment:
This comment suggests that evaluation should be, no shell **or** username, however you use AND operand in the code? I'm commenting to avoid potential future bugs and for those who maybe interested in producing future PRs

Reply:
Looking at the API (https://dlang.org/library/std/path/expand_tilde.html) of `expandTilde` it seems that only `HOME` is used:

> There are two ways of using tilde expansion in a path. One involves using the tilde alone or followed by a path separator. In this case, the tilde will be expanded with the value of the environment variable HOME. 

So I am not sure whether the check for SHELL is necessary at all.

Reply:
Checking for SHELL & USER was the most reliable way to get what the environment was doing on CentOS under systemd / cron operation. Without checking SHELL as well, the application under CentOS would not correctly determine what '~' should be expanded to.